### PR TITLE
Port mongodb@2.6 from versions repo

### PR DESCRIPTION
**Pull request by jeffheifetz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jeffheifetz** [comment]
Ported https://github.com/Homebrew/homebrew-versions/pull/1510

File: Formula/mongodb@2.6.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
+class MongodbAT26 < Formula
+  desc "High-performance document-oriented database"
+  homepage "https://www.mongodb.org/"
+  url "https://fastdl.mongodb.org/src/mongodb-src-r2.6.12.tar.gz"
+  sha256 "2dd51eabcfcd133573be74c0131c85b67764042833e7d94077e86adc0b9406dc"
+
+  option "with-boost", "Compile using installed boost, not the version shipped with mongodb"
```
Comment (MikeMcQuaid) [inline Formula/mongodb@2.6.rb]: Let's remove this option.

File: Formula/mongodb@2.6.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  depends_on :macos => :snow_leopard
+  depends_on "scons" => :build
+  depends_on "boost" => :optional
+  depends_on "openssl" => :optional
```
Comment (MikeMcQuaid) [inline Formula/mongodb@2.6.rb]: What happens if this isn't enabled?

Comment (jeffheifetz) [inline Formula/mongodb@2.6.rb]: I honestly don't know

Comment (jeffheifetz) [inline Formula/mongodb@2.6.rb]: It still worked when I tested it locally

File: Formula/mongodb@2.6.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
+class MongodbAT26 < Formula
+  desc "High-performance document-oriented database"
+  homepage "https://www.mongodb.org/"
+  url "https://fastdl.mongodb.org/src/mongodb-src-r2.6.12.tar.gz"
+  sha256 "2dd51eabcfcd133573be74c0131c85b67764042833e7d94077e86adc0b9406dc"
+
+  depends_on :macos => :snow_leopard
+  depends_on "scons" => :build
+  depends_on "boost" => :optional
+  depends_on "openssl" => :optional
```
Comment (MikeMcQuaid) [inline Formula/mongodb@2.6.rb]: Let's make this always-on (remove the `:optional`).

File: Formula/mongodb@2.6.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
+class MongodbAT26 < Formula
+  desc "High-performance document-oriented database"
+  homepage "https://www.mongodb.org/"
+  url "https://fastdl.mongodb.org/src/mongodb-src-r2.6.12.tar.gz"
+  sha256 "2dd51eabcfcd133573be74c0131c85b67764042833e7d94077e86adc0b9406dc"
+
+  depends_on :macos => :snow_leopard
+  depends_on "scons" => :build
+  depends_on "boost" => :optional
```
Comment (MikeMcQuaid) [inline Formula/mongodb@2.6.rb]: Remove this line.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
Couple more changes and this should be good to 🚢

File: Formula/mongodb@2.6.rb (hunk 4 above)
Comment (jeffheifetz) [inline Formula/mongodb@2.6.rb]: Sorry should have removed this the first time

**Comment by jeffheifetz** [comment]
Thanks so much @MikeMcQuaid

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**